### PR TITLE
CB-5546 Mock integration tests for parent-child environment relation

### PR DESCRIPTION
**Pull request by pkedvessy**
CB-5546
Mock integration tests for parent-child environment relation

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentTest.java [hunk 1]
```diff
@@ -157,17 +162,155 @@ public void testWlClusterWithInternalGetRequest(MockedTestContext testContext) {
...
+                .given(CredentialTestDto.class)
+                .when(credentialTestClient.create())
+                .given(PARENT_ENVIRONMENT, EnvironmentTestDto.class)
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .given(CHILD_ENVIRONMENT, EnvironmentTestDto.class)
+                .withParentEnvironmentName(testContext.get(PARENT_ENVIRONMENT).getName())
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .when(environmentTestClient.list())
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentTest.java]: What is the point with this line? If it has a purpose shouldn't it be checked?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/environment/EnvironmentTestDto.java [hunk 2]
```diff
@@ -202,6 +210,16 @@ public EnvironmentTestDto withS3Guard() {
         return this;
     }
 
+    public EnvironmentTestDto withParentEnvironmentName(String parentEnvironmentName) {
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/environment/EnvironmentTestDto.java]: it would make sense to implement this method with RunninParameter argument and let the method gain the environment name from the testcontext
withParentEnvironmentName(RunningParameter parentEnvKey) {
  getRequest().setEnvironmentName(resource.get(parentEnvKey.getKey());
...

but it is just nice to have

File: integration-test/src/main/resources/application.yml [hunk 3]
```diff
@@ -83,8 +83,8 @@ integrationtest:
             - http://s3.amazonaws.com/dev.hortonworks.com/CFM/centos7/2.x/BUILDS/2.0.0.0-22/tars/parcel/NIFIREGISTRY-0.5.0.2.0.0.0-22.jar
 
   user:
-    accessKey: Y3JuOmFsdHVzOmlhbTp1cy13ZXN0LTE6Y2xvdWRlcmE6dXNlcjptb2NrdXNlcg==
-    secretKey: nHkdxgZR0BaNHaSYM3ooS6rIlpV5E+k1CIkr+jFId2g=
+    accesskey: Y3JuOmFsdHVzOmlhbTp1cy13ZXN0LTE6Y2xvdWRlcmE6dXNlcjptb2NrdXNlcg==
+    secretkey: nHkdxgZR0BaNHaSYM3ooS6rIlpV5E+k1CIkr+jFId2g=
```
Comment (afarsang) [inline integration-test/src/main/resources/application.yml]: Are these changes necessary? Seems unrelated

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/FreeIpaAttachChildEnvironmentTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        testContext
+                .given(FreeIPATestDto.class).withCatalog(testContext.getImageCatalogMockServerSetup().getFreeIpaImageCatalogUrl())
+                .when(freeIPATestClient.create())
+                .await(Status.AVAILABLE)
+                .given(FreeIPAChildEnvironmentTestDto.CHILD_ENVIRONMENT_KEY, EnvironmentTestDto.class).withCreateFreeIpa(false)
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .given(FreeIPAChildEnvironmentTestDto.class)
+                .when(freeIPATestClient.attachChildEnvironment())
+                .then(FreeIpaChildEnvironmentAssertion.validate())
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/FreeIpaAttachChildEnvironmentTest.java]: As I recall we planned to check freeipa mock calls as well, don't we?

File: integration-test/src/main/resources/application.yml (hunk 3 above)
Comment (Bajzathd) [reply integration-test/src/main/resources/application.yml]: They are unrelated, but when I mistakenly tried to run the tests on AWS, I found this typo - the tests would not start because these properties are referenced lowercase in the code

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentChildTest.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                .when(environmentTestClient.list())
+                .then(this::checkEnvIsListedByNameAndParentName)
+                .given(CHILD_ENVIRONMENT, EnvironmentTestDto.class)
+                    .withParentEnvironment()
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .when(environmentTestClient.list())
+                .then(this::checkEnvIsListedByNameAndParentName)
+                .then((testContext1, testDto, client) -> {
+                    Mockito.verify(dnsZoneAddResponse).handle(any(), any());
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentChildTest.java]: Although it is working, this is not the way what we use to follow in our mock tests. We are using a httpmock (for FreeIPA as well), and we are use to seek the http mock calls for validation. You can see examples in DistroXClusterCreationTest

**Comment by Bajzathd** [comment]
retest this please
